aqt/installer.py

```diff
@@ -646,12 +706,68 @@ def _create_parser(self):
+        # parser.set_defaults(func=self.show_help)
+
+        # list2: placeholder name for added functionality; don't know what to call it!
+        list_parser = subparsers.add_parser(
+            "list",
+            formatter_class=argparse.RawTextHelpFormatter,
+            epilog="Examples:\n"
+            "$ aqt list qt5 windows desktop --filter-minor 9                   # print all versions of Qt 5.9\n"
+            "$ aqt list qt5 windows desktop --filter-minor 9 --latest-version  # print latest Qt 5.9\n"
+            "$ aqt list qt5 windows desktop --filter-minor 9 --latest-packages # print packages for latest 5.9\n"
+            "$ aqt list qt5 windows desktop --list-packages 5.12.0             # print packages for 5.12.0\n",
+        )
         list_parser.set_defaults(func=self.run_list)
-        self._set_common_argument(list_parser)
-        help_parser = subparsers.add_parser("help")
-        help_parser.set_defaults(func=self.show_help)
-        parser.set_defaults(func=self.show_help)
+        list_parser.add_argument(
+            "category",
+            choices=["tools", "qt5", "qt6"],
+            help="category of packages to list",
+        )
+        list_parser.add_argument(
+            "host", choices=["linux", "mac", "windows"], help="host os name"
+        )
+        list_parser.add_argument(
+            "target", choices=["desktop", "winrt", "android", "ios"], help="target sdk"
+        )
+        list_parser.add_argument(
+            "--extension",
+            choices=ALL_EXTENSIONS,
+            help="extension of packages to list",
+        )
+        # list_parser.add_argument("qt_version", required=False, type=str,
+        #                          help='Qt version in the format of "5.X.Y". If this is set, then ')
+        list_parser.add_argument(
+            "--filter-minor",
+            type=int,
+            metavar="MINOR_VERSION",
+            help="print versions for a particular minor version. "
+            "IE: `aqt list qt5 windows desktop --filter-minor 12` prints all versions beginning with 5.12",
+        )
+        output_modifier_exclusive_group = list_parser.add_mutually_exclusive_group()
+        output_modifier_exclusive_group.add_argument(
+            "--list-packages",
+            type=str,
+            metavar="VERSION",
+            help='Qt version in the format of "5.X.Y". '
+            "When set, this lists all the packages available for Qt 5.X.Y.",
+        )
+        output_modifier_exclusive_group.add_argument(
+            "--latest-version",
+            action="store_true",
+            help="print only the newest version available",
+        )
+        output_modifier_exclusive_group.add_argument(
+            "--latest-packages",
+            action="store_true",
+            help="list all the packages available for the latest version of Qt, "
+            "or a minor version if the `--filter-minor` flag is set.",
+        )
```

Comment:
These two options, along with the `--filter-minor` option, are intended to help people who might be using the output of `aqt list` in a shell script, and don't want to have to filter the output manually. I can take them out if necessary.